File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.js
/**
 * @param {string[]} folder
 * @return {string[]}
 */
var removeSubfolders = function (folder) {
    folder.sort();
    const result = [];

    for (const f of folder) {
        if (result.length === 0 || !f.startsWith(result[result.length - 1] + '/'))
            result.push(f);
    }

    return result;
};

File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.py
class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        n = len(folder)
        folder.sort()
        list = []
        list.append(folder[0])

        for i in range(1, n):
            last = list[-1]
            if not folder[i].startswith(last + "/"):
                list.append(folder[i])
        
        return list